### PR TITLE
DATACOUCH-574 - Support countBy...() query method calls in repository.

### PR DESCRIPTION
**Pull request by mikereiche**
- [x] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [x] There is a ticket in the bug tracker for the project in our [JIRA](https://jira.spring.io/browse/DATACOUCH).
- [x] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

### CODE REVIEW

**Comment by mp911de** [comment]
Please mind the commit message conventions:

The commit message should follow the following style:

> * First line starts with the ticket id.
> * Separate ticket id and summary with a dash.
> * Start summary with a capital letter.
> * Finish summary with a dot.


See https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc#commit-messages for further reference.

File: src/main/java/org/springframework/data/couchbase/repository/query/CouchbaseQueryMethod.java [hunk 1]
```diff
@@ -163,6 +163,16 @@ public String getInlineN1qlQuery() {
...
 	}
 
+	/**
+	 * indicates if the method begins with "count"
+	 *
+	 * @return true if the method begins with "count", indicating that .count() should be called instead of one() or
+	 *         all().
+	 */
+	public boolean isCountQuery() {
+		return getName().startsWith("count");
```
Comment (daschl) [inline src/main/java/org/springframework/data/couchbase/repository/query/CouchbaseQueryMethod.java:173]: could this be a case insensitive check as well to minimize the risk of not catching it?

File: src/test/java/org/springframework/data/couchbase/domain/AirportRepository.java [hunk 2]
```diff
@@ -37,4 +43,9 @@
 	@Query("#{#n1ql.selectEntity} where iata = $1")
 	List<Airport> getAllByIata(String iata);
 
+	long countByIataIn( String... iata );
```
Comment (daschl) [inline src/test/java/org/springframework/data/couchbase/domain/AirportRepository.java:46]: formatting: whitespace looks off inside the ()

File: src/test/java/org/springframework/data/couchbase/domain/AirportRepository.java [hunk 3]
```diff
@@ -37,4 +43,9 @@
...
+
+	long countByIcaoAndIataIn(String icao,  String... iata );
```
Comment (daschl) [inline src/test/java/org/springframework/data/couchbase/domain/AirportRepository.java:48]: here as well

File: src/main/java/org/springframework/data/couchbase/repository/query/CouchbaseQueryMethod.java (hunk 1 above)
Comment (mikereiche) [reply src/main/java/org/springframework/data/couchbase/repository/query/CouchbaseQueryMethod.java]: Ok, good idea.

File: src/test/java/org/springframework/data/couchbase/domain/AirportRepository.java (hunk 2 above)
Comment (mikereiche) [reply src/test/java/org/springframework/data/couchbase/domain/AirportRepository.java]: ok.

File: src/test/java/org/springframework/data/couchbase/domain/AirportRepository.java (hunk 3 above)
Comment (mikereiche) [reply src/test/java/org/springframework/data/couchbase/domain/AirportRepository.java]: ok

**Pull request closed without merging**